### PR TITLE
Pretty print CNI config when rendering NetAttachDefs

### PR DESCRIPTION
**Pull request by tariq1890**
When debugging CNI configs, it becomes a lot easier when we can see pretty-printed JSON in NetworkAttachmentDefinitions

### CODE REVIEW

**Comment by tariq1890** [comment]
@adrianchiris @SchSeba @e0ne 

Request review on this as well

File: api/v1/helper.go
Comment (adrianchiris) [inline api/v1/helper.go:524]: just wondering if this will make reading logs better or worse when they are scraped and sent over to a centralized logging and viewed from a web console e.g graylog / ELK stack or similar.

Comment (tariq1890) [reply api/v1/helper.go:524]: hmmm, I hadn't thought of that. I know SumoLogic has auto-formatting capabilities for JSON. So the pretty-printing would be idempotent. Not sure of other tools

Comment (tariq1890) [reply api/v1/helper.go:524]: Just to be clear, this change is so that we can view the plugins config well-formatted in the NetAttachDef CR

**Comment by tariq1890** [comment]
@adrianchiris Any update on this?

**Comment by tariq1890** [comment]
@adrianchiris I have updated the PR to ensure the rendered CNI config in the NetAttachDef is pretty-printed. Requesting your review on this

File: controllers/sriovibnetwork_controller.go
Comment (adrianchiris) [inline controllers/sriovibnetwork_controller.go:133]: nit: could you move this to after L119  so its in the same place where netAttDef is created.

File: controllers/sriovnetwork_controller.go
Comment (adrianchiris) [inline controllers/sriovnetwork_controller.go:134]: same comment as above

Comment (tariq1890) [reply controllers/sriovnetwork_controller.go]: done. Thanks!

File: controllers/sriovibnetwork_controller.go
Comment (tariq1890) [reply controllers/sriovibnetwork_controller.go]: done. Thanks!

**Review by adrianchiris — APPROVED** [review]

File: test/util/util.go
Comment (tariq1890) [reply test/util/util.go:275]: I thought panic was justified here, as

1) as the input to this function is deterministic, we are only passing in static json strings that we deem as expected output

2) This function is defined in the test package

Comment (zeeke) [reply test/util/util.go:275]: Well, the function is defined in the `test/util` package and it's called by exported functions. Actually, these functions are invoked only by  `controllers/*_test.go` files but `test/util/*` packages are used as a test library from other projects.

If you make the whole chain of function calls package-private, it looks good to me to raise a panic.

Comment (tariq1890) [reply test/util/util.go]: Hi @zeeke I've just refactored the code to make it all package private. Can you check now?

**Review by zeeke — APPROVED** [review]
/lgtm 

@tariq1890 thanks for working on this

waiting for @adrianchiris approval as the code changed a bit since his last review

File: controllers/helper.go
Comment (adrianchiris) [inline controllers/helper.go:53]: why was this method moved ? as its only used in tests it should stay where it was IMO.

File: controllers/sriovibnetwork_controller_test.go
Comment (adrianchiris) [inline controllers/sriovibnetwork_controller_test.go:194]: why was it moved from test/utils ?

@fedepaol  @SchSeba do you know if test/utils module being used in other project ?

File: controllers/sriovnetwork_controller_test.go
Comment (adrianchiris) [inline controllers/sriovnetwork_controller_test.go:220]: Moving these methods around is not something that is directly related to this PR so im not a fan of mixing the two.
if they are mixed then at least it should be in separate commits.


also im not too sure if these methods are used externally from /test/util.go
@fedepaol @SchSeba

File: controllers/helper.go
Comment (tariq1890) [reply controllers/helper.go:53]: fair, I will move back this back into the test files

**Comment by tariq1890** [comment]
@adrianchiris I have split the commits into two. Please re-review

**Comment by adrianchiris** [comment]
lets get inputs from @SchSeba @fedepaol regarding the moved methods, i dont remember if these are used in a CI/testing proj in Openshift

**Comment by zeeke** [comment]
I can confirm that methods `GenerateExpectedNetConfig` and `GenerateExpectedIBNetConfig` are not used by Openshift testing project:
- https://github.com/openshift-kni/cnf-features-deploy/tree/master/cnf-tests/testsuites

I asked @tariq1890 to move them because they can panic since this PR, so it's better to break compile compatibility (if any)  than getting a panic inside a library.

**Comment by tariq1890** [comment]
Hello :) how can we move this PR forward? It's been open for a while and I've been through multiple iterations already. Hoping to close this out

**Comment by adrianchiris** [comment]
thx for confirming @zeeke. ill merge this one once CI pass.

**Pull request merged**
